### PR TITLE
Bump CAPI to v1.5.0

### LINKED ISSUE #4375: CAPI v1.5.0-beta.0 has been released and is ready for testing
CAPI `v1.5.0-beta.0` has been released and is ready for testing.
Looking forward to your feedback before CAPI 1.5.0 release, the 25th July 2023!

## For quick reference

- [CAPI v1.5.0-beta.0 release notes](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.5.0-beta.0)
- [Shortcut to CAPI git issues](https://github.com/kubernetes-sigs/cluster-api/issues)

## Following are the planned dates for the upcoming releases

|Release|Expected Date|
|-----|-----|
|v1.5.0-beta.x | Tuesday 5th July 2023|
|release-1.5 branch created (Begin [Code Freeze])|Tuesday 11th July 2023|
|v1.5.0-rc.0 released|Tuesday 11th July 2023|
| release-1.5 jobs created | Tuesday 11th July 2023 |
| v1.5.0-rc.x released | Tuesday 18th July 2023 |
| v1.5.0 released | Tuesday 25th July 2023 |

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What type of PR is this?**
/kind support

**What this PR does / why we need it**:
Bumps CAPI to v1.5.0-rc.1 based on https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/providers/migrations/v1.4-to-v1.5.md

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4375, #4390

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

**Release note**:

```release-note
Bump CAPI to v1.5.0
```

### CODE REVIEW

**Comment by richardcase** [comment]
@furkatgofurov7 - maybe a `make generate` is needed?

File: bootstrap/eks/controllers/eksconfig_controller.go [hunk 1]
```diff
@@ -312,7 +312,7 @@ func (r *EKSConfigReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Man
 	}
 
 	err = c.Watch(
-		&source.Kind{Type: &clusterv1.Cluster{}},
+		source.Kind(mgr.GetCache(), &clusterv1.Cluster{}),
```
Comment (furkatgofurov7) [inline bootstrap/eks/controllers/eksconfig_controller.go:315]: unit tests are failing due to this, passing a cache to source.Kind does not seem to be properly working and tests are failing due to the object is missing, looking into it

Comment (furkatgofurov7) [reply bootstrap/eks/controllers/eksconfig_controller.go:315]: Thanks @richardcase for helping to identify the issue with fakeclient and fixing them!

**Comment by furkatgofurov7** [comment]
/retitle Bump CAPI to v1.5.0-rc.1

This should be ready for review!

/cc @richardcase @Ankitasw @Skarlso

**Comment by furkatgofurov7** [comment]
`pull-cluster-api-provider-aws-e2e` is failing due to go bump, we should rerun the test once the [fix](https://github.com/kubernetes/test-infra/pull/30152) is merged in infra

**Comment by furkatgofurov7** [comment]
Retriggering the job, looks like a timeout to me at a first glance

/test pull-cluster-api-provider-aws-e2e

**Comment by richardcase** [comment]
@furkatgofurov7 could you be able to update this to use **v1.5.0** instead of the RC now it's released?

**Comment by furkatgofurov7** [comment]
> @furkatgofurov7 could you be able to update this to use **v1.5.0** instead of the RC now it's released?

@richardcase it has been already updated to v1.5.0 but anyways needs a rebase, pushing it..

**Comment by richardcase** [comment]
Re-running it case it was a flake:

/test pull-cluster-api-provider-aws-e2e

**Comment by furkatgofurov7** [comment]
> 

GH rate limit was hit with this run, rerunning it again

/test pull-cluster-api-provider-aws-e2e

**Comment by furkatgofurov7** [comment]
> GH rate limit was hit with this run, rerunning it again

same

GH rate limit was hit with this run, rerunning it again

**Comment by furkatgofurov7** [comment]
> > Re-running it case it was a flake:
> 
> GH rate limit was hit with this run, rerunning it again


same

 /test pull-cluster-api-provider-aws-e2e

**Comment by Ankitasw** [comment]
@furkatgofurov7 do you need any help here?

**Comment by Ankitasw** [comment]
@richardcase would you be able to spend some time here?

**Comment by Ankitasw** [comment]
@furkatgofurov7 @richardcase I dont have permission to edit the PR so couldn't push the changes, could any of you resolve the conflicts, then we can merge this on priority

**Comment by Ankitasw** [comment]
I ll cherry pick the changes from this PR and raise a new one for fixing the tests and rebasing.

**Comment by Ankitasw** [comment]
/close

in favor of #4510

**Pull request closed without merging**